packages/core/src/useNavigationBuilder.tsx

```diff
@@ -254,6 +258,16 @@ export default function useNavigationBuilder<
     focusedListeners,
   });
 
+  const getStateForRoute = React.useCallback(
```

Comment:
Move this to `useStateGetters` hook, and return this instead of exposing the `stateGetters` ref